x-pack/legacy/plugins/siem/public/components/edit_data_provider/index.tsx

```diff
@@ -88,37 +82,81 @@ export const getInitialOperatorLabel = (
   }
 };
 
-export class StatefulEditDataProvider extends React.PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
+export const StatefulEditDataProvider = React.memo<Props>(
+  ({
+    andProviderId,
+    browserFields,
+    onDataProviderEdited,
+    providerId,
+    timelineId,
+    field,
+    isExcluded,
+    operator,
+    value,
+  }) => {
+    const [updatedField, setUpdatedField] = useState([
+      { label: field },
+    ] as EuiComboBoxOptionProps[]);
+    const [updatedOperator, setUpdatedOperator] = useState(getInitialOperatorLabel(
+      isExcluded,
+      operator
+    ) as EuiComboBoxOptionProps[]);
+    const [updatedValue, setUpdatedValue] = useState(value as string | number);
+
+    /** Focuses the Value input if it is visible, falling back to the Save button if it's not */
+    const focusInput = () => {
```

Comment:
General comment, I will avoid lambda function inside a component and do that
```
 /** Focuses the Value input if it is visible, falling back to the Save button if it's not */
    function focusInput() {
      const elements = document.getElementsByClassName(VALUE_INPUT_CLASS_NAME);

      if (elements.length > 0) {
        (elements[0] as HTMLElement).focus(); // this cast is required because focus() does not exist on every `Element` returned by `getElementsByClassName`
      } else {
        const saveElements = document.getElementsByClassName(SAVE_CLASS_NAME);

        if (saveElements.length > 0) {
          (saveElements[0] as HTMLElement).focus();
        }
      }
    }
```

Reply:
And do the same thing for the other lambda